### PR TITLE
improve Publisher,Subscribe

### PR DESCRIPTION
**Pull request by kongliang1**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+	return t
+}
+
+func (t *Client) MoodType(typeName string) *Client {
+	t.message.MoodType = typeName
+	return t
+}
+
+func (t *Client) Priority(priority float64) *Client {
+	if priority > 1000 {
```
Comment (zhongdalu-trechina) [inline client.go:114]: `>=1000` or `>999`

Comment (kongliang1) [reply client.go:114]: ok, will fix it.

File: redis_schedule_job.go [hunk 2]
```diff
@@ -232,19 +232,19 @@ func (t *scheduleJob) doConsumeZset(ctx context.Context, vals []string, consumer
 func (t *scheduleJob) sendToStream(ctx context.Context, msg *Message) error {
 	mmsg := messageToMap(msg)
 	subType := normalSubscribe
-	if msg.MoodType == "sequential" {
+	if msg.MoodType == string(SEQUENTIAL) {
```
Comment (zhongdalu-trechina) [inline redis_schedule_job.go:235]: no need `string(...)`

File: client.go [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+func (t *Client) Payload(payload []byte) *Client {
+	t.message.Payload = stringx.ByteToString(payload)
+	return t
+}
+
+func (t *Client) process(ctx context.Context, cmd IBaseCmd) error {
+
+	defer func() {
+		t.message = &Message{
```
Comment (zhongdalu-trechina) [inline client.go:129]: should not reset `t.message` here, it is not thread safe.

File: client.go [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+			MoodType:  string(NORMAL),
+		}
+	}()
+
+	if err := cmd.filter(t.message); err != nil {
+		return err
+	}
+
+	if cmd, ok := cmd.(*Publish); ok {
+		t.message.ExecuteTime = cmd.executeTime
```
Comment (zhongdalu-trechina) [inline client.go:147]: not safe too.

File: client.go (hunk 3 above)
Comment (kongliang1) [reply client.go:145]: done

File: client.go (hunk 4 above)
Comment (kongliang1) [reply client.go:165]: done

File: redis_schedule_job.go (hunk 2 above)
Comment (tanvir-retailai) [reply redis_schedule_job.go:234]: @kongliang1 did you fix this one?

Comment (kongliang1) [reply redis_schedule_job.go:234]: There is no need to make any modifications here.  because data type has changed.

**Pull request merged**